js/src/components/app.js

```diff
@@ -62,14 +62,20 @@ export default class App extends Component {
   }
   nextPageHandler = (e) => {
     e.preventDefault();
-    this.fetchLogEntries(this.state.page + 1);
+    this.setState({
+      page: this.state.page + 1
```

Comment:
Does not work, because setting the state is not exactly immediate

Reply:
In this case you can use ```this.setState((prevState, props) => ({
  counter: prevState.page + 1,
}));```

Reply:
I didn't try that, but one issue I had was the async nature. It executed the HTTP request before state settled. I updated to

```
    this.setState({
      page: this.state.page - 1
    }, () => {
      this.fetchLogEntries(this.state.page);
    });
```

Reply:
Oh i see. Yeah the above example is useful when the HTTP request would change the actual state.